File: java/Watcher/src/CurrentConditionsDisplay.java
import Interfaces.*;

public class CurrentConditionsDisplay implements IObserver, IDisplayElement {

	private float temperature;
	private float humidity;
	private ISubject weatherData;
	
	public CurrentConditionsDisplay(ISubject weatherData) {
		this.weatherData = weatherData;
		weatherData.registerObserver(this);
	}
	
	@Override
	public void display() {
		System.out.println("Current conditions:" + temperature +
				"F degrees and " + humidity + "% humidity");

	}

	@Override
	public void update(float temp, float humidity, float pressure) {
		this.temperature = temp;
		this.humidity = humidity;
		display();

	}

}

File: c#/Watcher/Models/CurrentConditionsDisplay.cs
using System;
using Watcher.Interfaces;

namespace Watcher.Models
{
    public class CurrentConditionsDisplay : IWeatherObserver
    {
        private float Temperature { get; set; }
        private float Humidity { get; set; }
        private IWeatherObservable WeatherData { get; set; }

        public CurrentConditionsDisplay(IWeatherObservable weatherData)
        {
            WeatherData = weatherData;
            weatherData.AddObserver(this);
        }

        public void Update(float temp, float humidity, float pressure)
        {
            Temperature = temp;
            Humidity = humidity;

            Console.WriteLine($"Current conditions: {Temperature}F degrees and {Humidity}% humidity");
        }
    }
}
